mir-module/src/main/resources/META-INF/resources/js/mir/xeditor-form.js

```diff
@@ -284,11 +317,16 @@ $(document).ready(function() {
      // Enables the datetimepicker
      if (jQuery.fn.datepicker) {
          $('.datetimepicker').find('input').each( function(index, elm){
-             pickDatePickerFormatAndAdd(elm);
+             if ($(elm).val().includes("T")) {
```

Comment:
Thank you for bringing this to my attention. The code you mentioned is located within the "$('.datetimepicker').find('input').each( function(index, elm){...}" block, and only executed when editing a document that already contains a date or datetime value in one of its input fields.

It is not intended to interfere with the creation of a new document but rather comes into play during the editing process. The purpose of this code is to recognize fields with datetime values and automatically select the "Date and Time" format upon loading the editor, instead of the standard "date" format. 

To use the timepicker when creating a document, you can click the grey arrow on the right side on the input field and select "Date and Time" to have the timepicker included. By default it is only set to "date" so you can not pick a time.
I hope this clears things up. If there are further questions please feel free to let me know.

Reply:
Sorry, but i need to disagree here.

1. A technicality: in your first paragraph, your write "The code you mentioned is [...] only executed when editing a document that already contains a date or datetime value in one of its input fields." That is not correct. `$('.datetimepicker').find('input').each(...);` specifically means that the code is executed for every input field marked with class `datetimepicker` in every form. Whether or not they already contain a datetime value is irrelevant. 
2. More importantly: you say  that "[i]t is not intended to interfere with the creation of a new document". That might be true for _the specific use case you have in mind_, but this component is a general component that is and will be used for many other use cases. The component should therefore behave in a consistent way.
    - As a user I'd expect the same capability (to select a time value in addition to a date value) independent of whether I create a new document or edit an existing one.
    - From a metadata point of view: if it makes sense for a form field to contain date and time information during editing, I'd suggest that this also makes sense during creation of a new document. 
3. Still more importantly and as alluded to in my previous comment: this component is already in use. Up until now it was only used to select date information. Supporting time information as well fundamentally changes the nature of this component. That is certainly a useful improvement, but some form of backwards compatibility to support the existing uses (where only date information is required and time information is neither needed nor wanted) **is a must**. If that requires a migration step, that would be fine for me, but the target branch needs to be changed then.

Lastly, I've checked out this PR in conjunction with the current 2023.06.1-SNAPSHOT version of MyCoRe, and the "grey arrow on the right side on the input field" you talk about isn't shown to me.

Example (form for a dissertation with two fields that have the class `datetimepicker`):
![image](https://github.com/MyCoRe-Org/mir/assets/6521030/82cc844a-6019-4355-9366-983fa4ef9100)